src/plugin/src/main/java/com/defenseunicorns/uds/keycloak/plugin/CustomAWSSAMLAttributeMapper.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2024 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+package com.defenseunicorns.uds.keycloak.plugin;
+
+import org.keycloak.dom.saml.v2.assertion.AttributeStatementType;
+import org.keycloak.dom.saml.v2.assertion.AttributeType;
+import org.keycloak.dom.saml.v2.assertion.AttributeStatementType.ASTChoiceType;
+import org.keycloak.models.AuthenticatedClientSessionModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.ProtocolMapperModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.models.UserSessionModel;
+import org.keycloak.models.utils.KeycloakModelUtils;
+import org.keycloak.protocol.saml.mappers.AbstractSAMLProtocolMapper;
+import org.keycloak.protocol.saml.mappers.AttributeStatementHelper;
+import org.keycloak.protocol.saml.mappers.SAMLAttributeStatementMapper;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.saml.common.constants.JBossSAMLURIConstants;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CustomAWSSAMLAttributeMapper extends AbstractSAMLProtocolMapper implements SAMLAttributeStatementMapper {
+
+    public static final String PROVIDER_ID = "aws-saml-attribute-mapper";
+    private static final String AWS_PRINCIPAL_TAG_PREFIX = "https://aws.amazon.com/SAML/Attributes/PrincipalTag:";
+
+    private static final List<ProviderConfigProperty> configProperties = new ArrayList<ProviderConfigProperty>();
+
+    static {
+        ProviderConfigProperty property;
+
+        property = new ProviderConfigProperty();
+        property.setName("user.attribute");
```

Comment:
Could you please extract these names into constants. It will be harder to make a typo this way.